components/support/utils/src/test/java/mozilla/components/support/utils/SafeIntentTest.kt

```diff
@@ -4,23 +4,284 @@
 
 package mozilla.components.support.utils
 
+
 import android.content.Intent
+import android.os.Parcelable
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
 import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
-import org.mockito.ArgumentMatchers
+import org.mockito.ArgumentMatchers.anyBoolean
+import org.mockito.ArgumentMatchers.anyInt
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.Mockito.doReturn
 import org.mockito.Mockito.doThrow
 import org.mockito.Mockito.mock
 import org.robolectric.RobolectricTestRunner
+import java.util.HashSet
+import kotlin.collections.ArrayList
 
 @RunWith(RobolectricTestRunner::class)
 class SafeIntentTest {
+
+    private lateinit var intent: Intent
+
+    @Before
+    fun setup() {
+        intent = mock(Intent::class.java)
+    }
+
     @Test
     fun `getStringArrayListExtra returns null if intent throws OutOfMemoryError`() {
-        val intent = mock(Intent::class.java)
+
         doThrow(OutOfMemoryError::class.java)
-                .`when`(intent).getStringArrayListExtra(ArgumentMatchers.anyString())
+                .`when`(intent).getStringArrayListExtra(anyString())
 
         assertNull(SafeIntent(intent).getStringArrayListExtra("mozilla"))
     }
+
+    @Test
+    fun `getExtras returns null if intent throws OutOfMemoryError`() {
+        
+        doThrow(OutOfMemoryError::class.java)
+                .`when`(intent).extras
+
+        assertNull(SafeIntent(intent).extras)
+    }
+
+
```

Comment:
\[ktlint\] WARNING: \[no\-consecutive\-blank\-lines\] Needless blank line\(s\) in column 1